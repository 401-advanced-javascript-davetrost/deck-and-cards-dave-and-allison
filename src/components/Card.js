import React from 'react';
import PropTypes from 'prop-types';

const Card = ({ _id, title, image, content }) => (
  <div key={_id}>
    <header><h3>{title}</h3></header>
    <figure>
      <img src={image} alt={title} />
    </figure>
    {content}
  </div>
);

Card.propTypes = {
  _id: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  image: PropTypes.string.isRequired,
  content: PropTypes.string
};

export default Card;
