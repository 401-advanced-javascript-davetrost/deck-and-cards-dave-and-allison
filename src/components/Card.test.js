import React from 'react';
import { shallow } from 'enzyme';
import Card from './Card';

describe('Card component', () => {
  it('renders with content', () => {
    const wrapper = shallow(
      <Card 
        title="Test Title"
        image="test image"
        content="this is a test" />
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('renders without content', () => {
    const wrapper = shallow(
      <Card 
        title="Test Title"
        image="test image" />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
