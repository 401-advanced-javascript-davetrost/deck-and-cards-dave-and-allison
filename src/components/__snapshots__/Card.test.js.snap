// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component renders with content 1`] = `
<div>
  <header>
    <h3>
      Test Title
    </h3>
  </header>
  <figure>
    <img
      alt="Test Title"
      src="test image"
    />
  </figure>
  <p>
    this is a test
  </p>
</div>
`;

exports[`Card component renders without content 1`] = `
<div>
  <header>
    <h3>
      Test Title
    </h3>
  </header>
  <figure>
    <img
      alt="Test Title"
      src="test image"
    />
  </figure>
</div>
`;
